keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1006 @@
+    used.
+  * It does not have annotation `resource.kubernetes.io/pod-claim-name:` as
+    it is created for the extended resource request(s) in a pod spec, not for a
+    claim in the pod spec.
+  * It does have annotation `resource.kubernetes.io/extended-resource-claim: pod-name` as
+    it is created, deleted, updated by the scheduler. It is used by scheduler
+    to find the resource claim it has created, and ensure at most one such
+    claim per pod.
+  * At most one such claim object is created per pod. For example, if a pod
+    requests for foo1.domain/bar and foo2.domain/bar, the allocation of devices
+    for each are recorded in DeviceResourceRequestAllocationResult, and just 
+    one claim object with allocation `Results` that lists all allocated devices
+    is created for the pod.
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic resource plugin during
+    preBind phase. The in-memory one in the assumed cache is created earlier
+    during Reserve phase.
+  * It is *deleted*
+    * either together with the owning pod's deletion.
+    * or by the scheduler dynamic resource plugin during unReserve phase.
+    * or by the scheduler dynamic resource plugin during postFilter phase.
+  * It is *discovered* by the kubelet via `pod.Status.ExtendedResourceClaimStatus`
+  * It is *read* by the kubelet DRA device driver to prepare the devices listed
+    therein when preparing to run the pod.
+
+```go
+type DeviceRequest struct {
+	// Name can be used to reference this request in a pod.spec.containers[].resources.claims
+	// entry and in a constraint of the claim.
+	//
+	// Must be a DNS label.
+	//
+	// +required
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
+}
+```
+
+To enable the kubelet to map devices back to the containers which requested them,
+the kube-scheduler creates one `DeviceRequest` per extended resource backed by DRA
+per container in the pod. The name of the `DeviceRequest` has the form
+"container-%d-request-%d", where the first %d is the index of the container in the pod.
+The second %d is the index of the extended resource inside the container
+resource requests. For example, if the first container in the pod has an
+extended resource backed by DRA which is the 3rd such request in the container,
+then the name of the `DeviceRequest` is "container-0-request-2".
+
+Documenting this naming is merely informational, it is not part of the API.
+The kubelet must not rely on it. Instead, the
+`ContainerExtendedResourceRequest` field below specifies the mapping.
+
+### Pod API
+
+A new field `extendedResourceClaimStatus` is added to Pod's status to track
```

Comment:
I like how well "Extended Resource" and "ResourceClaim" combine into "ExtendedResourceClaimStatus" here :smile: 

"ExtendedResourceDeviceClaim"... not so much.

@yliaog: this refers to some old discussions around renaming ResourceClaim to DeviceClaim.